style/search.js

```diff
@@ -0,0 +1,263 @@
+  let p = result.properties;
+  let type = mapResultToDescription(p.type, p.osm_key, p.osm_value).replaceAll(
+    "_",
+    " "
+  );
+
+  let addr = collapseArray([p.housenumber, p.street], " ");
+  let name = p.name;
+
+  if (name == undefined) {
+    name = addr;
+    addr = undefined;
+  }
+
+  let description = collapseArray(
+    [addr, p.city, p.county, p.state, p.country],
+    ", "
+  );
+
+  let item = document.createElement("div");
+  item.className = "gc-result-item";
+
+  let itemCategory = document.createElement("div");
+  itemCategory.className = "gc-result-category";
+  itemCategory.innerText = type;
+  let itemName = document.createElement("div");
+  itemName.className = "gc-result-name";
+  itemName.innerText = name;
+  let itemDescription = document.createElement("div");
+  itemDescription.className = "gc-result-description";
+  itemDescription.innerText = description;
+
+  item.appendChild(itemCategory);
+  item.appendChild(itemName);
+  item.appendChild(itemDescription);
+
+  return item;
+}
+
+function geocoderResponse(data) {
+  liveResults.innerHTML = "";
+
+  for (let i = 0; i < data.features.length; i++) {
+    resultGeometry[i] = data.features[i].properties.extent;
+    resultPoint[i] = data.features[i].geometry.coordinates;
+    let result = geocoderResultEntry(data.features[i]);
+    liveResults.appendChild(result);
+    result.onclick = (e) => goToResult(i);
+  }
+}
+
+function search(e) {
+  let queryTerm = e.target.value;
+  resultSelectIndex = -1;
+  if (queryTerm.length < 3) {
+    liveResults.innerHTML = "";
+    return;
+  }
+
+  let position = map.getCenter();
+  let searchQuery = new URL("https://photon.komoot.io/api/?limit=3");
+  console.log(map.getZoom());
+
+  //Low zoom: generate "generic" search results
+  //High zoom: generate location-specific search results
+  if (map.getZoom() > 5) {
+    searchQuery.searchParams.set("lat", position.lat);
+    searchQuery.searchParams.set("lon", position.lng);
+  }
+
+  searchQuery.searchParams.set("q", e.target.value);
+
+  doSearch(searchQuery);
+}
+
+async function doSearch(searchQuery) {
+  //Abort prior search in progress
+  lastSearchRequest?.abort();
+  const controller = new AbortController();
+  lastSearchRequest = controller;
+
+  try {
+    const response = await fetch(searchQuery, { signal: controller.signal });
+    const data = await response.json();
+
+    if (controller.signal.aborted) {
+      return;
+    }
+    geocoderResponse(data);
+  } catch (e) {
+    if (e instanceof DOMException) {
+      //Do nothing; this is normal when search result is aborted by an
+      //additional keystroke
+    }
+  }
+}
+
+function arrowNavigate(e) {
+  if (e.defaultPrevented) {
+    return; // Do nothing if event already handled
+  }
+
+  let priorIndex = resultSelectIndex;
+
+  switch (e.code) {
+    case "ArrowUp":
+      // up arrow
+      resultSelectIndex--;
+      break;
+    case "ArrowDown":
+      // down arrow
+      resultSelectIndex++;
+      break;
```

Comment:
Prevent up/down going to beginning/end of text box.

```suggestion
    case "ArrowUp":
      // up arrow
      resultSelectIndex--;
      e.preventDefault();
      break;
    case "ArrowDown":
      // down arrow
      resultSelectIndex++;
      e.preventDefault();
      break;
```